run_tests.py

```diff
@@ -168,7 +171,75 @@ def print_report(total: set, failures_set: set, skipped_set: set):
 if len(expansion_failures) > 0:
     test_success = False
 
-print_report(tests, expansion_failures, expansion_skipped)
+print_report(expansion_tests, expansion_failures, expansion_skipped)
+
+print("\n* Variable parse tests")
+
+var_parse_tests = os.listdir(TEST_VAR_PARSE_PATH)
+var_parse_tests = [test for test in var_parse_tests if test.endswith(".env")]
+var_parse_tests.sort()
+
+var_parse_failures = set()
+var_parse_skipped = set()
+
+for test_name in var_parse_tests:
+    bash_version = (5, 0, 17) if "old" in test_name else None # else latest
+    test = os.path.join(TEST_VAR_PARSE_PATH, test_name)
+    logging.info(f'Test: {test_name}')
+
+    skip_test = test_name.startswith("skip")
+    if skip_test:
+        logging.info(f'Skipping...')
+        var_parse_skipped.add(test_name)
+        continue
+
+    expected_var_file = os.path.join(TEST_VAR_PARSE_PATH, test_name.replace(".env","_expected.py"))
+    expected_success = os.path.exists(expected_var_file)
+
+    expected = None
+    if expected_success:
+        with open(expected_var_file) as f:
+            expected = ast.literal_eval(f.read())
```

Comment:
Do we need literal_eval here? It seems like a bit of an overkill :) Is there any way to avoid evaluating arbitrary python code?

Reply:
I changed it to read in JSON data, instead of a python dictionary (with the only loss being aesthetic consistency).